public/app/features/commandPalette/actions/utils.ts

```diff
@@ -0,0 +1,144 @@
+import { useObservable } from 'react-use';
+import { Observable } from 'rxjs';
+
+import { ScopeNode } from '@grafana/data';
+import { t } from '@grafana/i18n';
+
+import { useScopesServices } from '../../scopes/ScopesContextProvider';
+import { ScopesSelectorServiceState } from '../../scopes/selector/ScopesSelectorService';
+import { NodesMap, SelectedScope, TreeNode } from '../../scopes/selector/types';
+import { CommandPaletteAction } from '../types';
+import { SCOPES_PRIORITY } from '../values';
+
+export function useScopeServicesState() {
```

Comment:
Wrapped the scope services access here so it does not need to be weirdly checked every time with conditional hook calls. Also make this easier to mock in the tests.